suite/tests/CMakeLists.txt

```diff
@@ -4044,6 +4036,19 @@ if (BUILD_CLIENTS)
         "${prefetch_analyzer_path}@-trace_dir@${testname_full}.*.dir/trace")
     endif ()
 
+    # Stress-test drcachesim with drstatecmp checks.
+    if (AARCH64)
+      torunonly_drcachesim(drstatecmp-fuzz drstatecmp-fuzz-app
+        "-trace_after_instrs 10M -enable_drstatecmp" "" "")
+    endif ()
+    if (NOT WIN32)
+      torunonly_drcachesim(drstatecmp-delay-simple ${ci_shared_app}
+        "-trace_after_instrs 20000 -exit_after_tracing 10000 -enable_drstatecmp" "")
+      torunonly_drcacheoff(drstatecmp-delay-simple ${ci_shared_app}
```

Comment:
How does this new test hit the app2app-truncation case? drmemtrace would truncate bbs in app2app only for scatter-gather instrs, right?

Reply:
It truncates rep strings.

Reply:
I.e., it truncates the block leading up to the rep string to isolate the rep string to its own bb, and truncates that as well.

Reply:
But ci_shared_app (suite/tests/client-interface/simple_app.c) is just a simple "hello world" program. Or does it contain repstr?

Reply:
hello,world runs over 100K instructions in ld.so and crt setup and is very likely to have multiple repstrs in there.  I guess it's not guaranteed if on a platform where the mem* and str* routine ifuncs resolve to an xmm-based scheme.